rotkehlchen/chain/solana/utils.py

```diff
@@ -1,4 +1,128 @@
+    """Unpacks the mint data from calling getAccountInfo on a token's mint address.
+    Full structure is as follows:
+    * Main mint info (82 bytes) parsed via MINT_LAYOUT:
+        - u32 mintAuthorityOption (4 bytes)
+        - publicKey mintAuthority (32 bytes)
+        - u64 supply (8 bytes)
+        - u8 decimals (1 byte)
+        - bool isInitialized (1 byte)
+        - u32 freezeAuthorityOption (4 bytes)
+        - publicKey freezeAuthority (32 bytes)
+    * Empty/optional data (83 bytes) - total 165 bytes (ACCOUNT_SIZE)
+    * Account type (1 byte)
+    * Extensions data (variable size) in TLV (type-length-value) format
+    https://github.com/solana-program/token-2022/blob/main/clients/js-legacy/src/state/mint.ts#L52
+    """
+    if len(mint_data) < MINT_SIZE:
+        return None
+
+    parsed_mint_data = MINT_LAYOUT.parse(mint_data)
+    return MintInfo(
+        supply=parsed_mint_data.supply,
+        decimals=parsed_mint_data.decimals,
+        tlv_data=(
+            mint_data[ACCOUNT_SIZE + ACCOUNT_TYPE_SIZE:]
+            if len(mint_data) > ACCOUNT_SIZE and mint_data[ACCOUNT_SIZE] == AccountType.Mint.value
+            else None
+        ),
+    )
+
+
+def get_extension_data(extension_type: ExtensionType, tlv_data: bytes) -> bytes | None:
+    """Gets the raw data for the specified extension from the TLV (Type Length Value) data.
+    TLV data is in the following format:
+    * extension type (2 bytes)
+    * length (2 bytes)
+    * data (length bytes)
+    """
+    offset = 0
+    while (data_offset := offset + 4) < len(tlv_data):
+        ext_type = int.from_bytes(tlv_data[offset:offset + 2], byteorder='little')
+        data_len = int.from_bytes(tlv_data[offset + 2:offset + 4], byteorder='little')
+        if ext_type == extension_type.value:
+            return tlv_data[data_offset:data_offset + data_len]
+        offset = data_offset + data_len
+
+    return None
+
+
+def decode_metadata_pointer(data: bytes) -> SolanaAddress | None:
+    """Decodes the metadata pointer extension data from the following structure:
+    * pubkey authority (32 bytes)
+    * pubkey metadata_address (32 bytes)
+    https://github.com/solana-program/token-2022/blob/main/interface/src/extension/metadata_pointer/mod.rs
+    Returns the metadata address.
+    """
+    if len(data) < 64:
+        return None
+
+    return SolanaAddress(str(Pubkey.from_bytes(data[32:64])))
+
+
+def decode_token_metadata(data: bytes, layout: Struct) -> tuple[str, str] | None:
+    """Decodes raw token metadata using the specified layout.
+    Returns a tuple with the name and symbol or None if the data is invalid.
+    TODO: Maybe include raw_metadata.uri and use it to query the token image as well?
+    """
+    try:
+        raw_metadata = layout.parse(data)
+        return (
+            raw_metadata.name.decode('utf-8').strip('\x00'),
+            raw_metadata.symbol.decode('utf-8').strip('\x00'),
+        )
```

Comment:
```suggestion
        return (
            (raw_metadata := layout.parse(data)).name.decode('utf-8').strip('\x00'),
            raw_metadata.symbol.decode('utf-8').strip('\x00'),
        )
```

also, why not make this a named tuple like we have for `MintInfo`